File: multiple languages/144. Binary Tree Preorder Traversal/main.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func preorderTraversal(root *TreeNode) []int {
	res := make([]int, 0)

	res = dfs(root, res)
	return res
}

func dfs(node *TreeNode, res []int) []int {
	if node == nil {
		return res
	}
	res = append(res, node.Val)
	res = dfs(node.Left, res)
	res = dfs(node.Right, res)
	return res
}

File: multiple languages/144. Binary Tree Preorder Traversal/main.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

 function preorderTraversal(root: TreeNode | null): number[] {
    const output = new Array();
    dfs(root, output);
    return output
};

const dfs = (node: TreeNode, output: number[]) => {
    if (node === null){
        return ;
    }

    output.push(node.val)
    dfs(node.left,output);
    dfs(node.right,output);

};